pkg/guardian/handler.go

```diff
@@ -10,12 +10,26 @@ const RequestsRemainingMax = math.MaxUint32
 // RequestBlockerFunc is a function that evaluates a given request and determines if it should be blocked or not and how many requests are remaining.
 type RequestBlockerFunc func(context.Context, Request) (bool, uint32, error)
 
+// CondChainResponse is a type that informs the condition chain when to stop processing the request, and if the request should be blocked or not.
+type RequestBlockerResp string
```

Comment:
Went with a string type (as opposed to some integer type and `iota`) because the string values are useful when debugging and testing.

Reply:
It probably wouldn't show up in benchmarks, but equality testing of an integer probably compiles to a single assembly instruction (or a few, at least), versus string equality which likely iterates through the backing bytes.

Stringer would satisfy your debug/testing uses. https://godoc.org/golang.org/x/tools/cmd/stringer

Reply:
Agreed, Stringer is a better choice in this case. Made this update.